mlflow/environment_variables.py

```diff
@@ -887,3 +887,16 @@ def get(self):
 MLFLOW_ENABLE_THREAD_LOCAL_TRACING_DESTINATION = _BooleanEnvironmentVariable(
     "MLFLOW_ENABLE_THREAD_LOCAL_TRACING_DESTINATION", False
 )
+
+#: Whether to allow pre-release versions when installing dependencies in UV-managed
+#: virtual environments created by MLflow. When set to True, UV will use the
+#: --prerelease=allow flag during pip install operations, which may install alpha,
+#: beta, or release candidate versions of packages.
+#: NOTE: Pre-release versions are also automatically allowed when MLFLOW_TESTING is set,
+#: to support testing with development versions of MLflow and its dependencies.
+#: WARNING: Pre-release versions may contain breaking changes and are not recommended
+#: for production use.
+#: (default: ``False``)
+MLFLOW_VIRTUALENV_UV_ALLOW_PRERELEASE = _BooleanEnvironmentVariable(
+    "MLFLOW_VIRTUALENV_UV_ALLOW_PRERELEASE", False
+)
```

Comment:
do we need this environment variables? Can users set `[UV_PRERELEASE](https://docs.astral.sh/uv/reference/environment/#uv_prerelease)`?

Reply:
The original intention was to allow users to 'kick the tires' by using mlflow's RC on logged models. I do think it's beneficial to provide that support and having an option to set that permits installing latest builds seems quite helpful for validation purposes during model development and testing.

Reply:
uv throws this message error for pre releases. I think the ideal for users is this message containing `UV_PRELEASE=allow`:

```python
 uv run --no-project --with mlflow==3.2.0rc0 mlflow --version
  × No solution found when resolving `--with` dependencies:
  ╰─▶ Because there is no version of mlflow-skinny==3.2.0rc0 and mlflow==3.2.0rc0 depends
      on mlflow-skinny==3.2.0rc0, we can conclude that mlflow==3.2.0rc0 cannot be used.
      And because you require mlflow==3.2.0rc0, we can conclude that your requirements
      are unsatisfiable.

      hint: `mlflow-skinny` was requested with a pre-release marker (e.g.,
      mlflow-skinny==3.2.0rc0), but pre-releases weren't enabled (try:
      `--prerelease=allow`)
```